### ISSUE TITLE
metricsProvider service port does not get registered with v1beta1.external.metrics.k8s.io inside cluster

### ISSUE THREAD

**Issue by omriarieli** [DataDog/helm-charts#30]
**Describe what happened:**
when installing the helm chart with values.yaml and changing metricsProvider.service.port to something other than the default (443) the external metrics provider registration of the cluster agent does not get updated with the correct port

**Describe what you expected:**
the APIService v1beta1.external.metrics.k8s.io will be updated accordingly

**Steps to reproduce the issue:**
change metricsProvider.service.port to something other than default e.g 1555
install the helm chart on the cluster
no external metrics are getting received from cluster-agent
`kubectl get APIService` results in:
`v1beta1.external.metrics.k8s.io        default/datadog-cluster-agent-metrics-api   False (ServicePortError)`

**Additional environment details (Operating System, Cloud provider, etc):**
using chart version 2.3.10, but i have noticed this since at least chart version 2.2.9. kubernetes version 1.15-17
this results in us having to manually edit the APIService each new installation and having to remember to do that or else our hpa based on custom metrics does not work

`kubectl get APIService v1beta1.external.metrics.k8s.io -oyaml`

```apiVersion: apiregistration.k8s.io/v1
kind: APIService
metadata:
  creationTimestamp: "2020-08-20T07:52:23Z"
  labels:
    app.kubernetes.io/instance: datadog
    app.kubernetes.io/managed-by: Helm
    app.kubernetes.io/name: datadog
    app.kubernetes.io/version: "7"
    helm.sh/chart: datadog-2.3.10
  name: v1beta1.external.metrics.k8s.io
  resourceVersion: "141754"
  selfLink: /apis/apiregistration.k8s.io/v1/apiservices/v1beta1.external.metrics.k8s.io
  uid: 4bfab8c3-567b-46e8-bc1f-ecf472b72eb9
spec:
  group: external.metrics.k8s.io
  groupPriorityMinimum: 100
  insecureSkipTLSVerify: true
  service:
    name: datadog-cluster-agent-metrics-api
    namespace: default
    port: 443
  version: v1beta1
  versionPriority: 100
status:
  conditions:
  - lastTransitionTime: "2020-08-20T07:52:23Z"
    message: service/datadog-cluster-agent-metrics-api in "default" is not listening
      on port 443
    reason: ServicePortError
    status: "False"
    type: Available```

**Comment by vboulineau** [comment]
Hi @omriarieli,

So your latest attempt was with `2.3.10`? Normally this issue was fixed in the `2.4.x` branch (so basically this repo).
Note that we check for Kubernetes version as it requires Kubernetes >= 1.15.

**Comment by omriarieli** [comment]
Hi @vboulineau thanks for the reply. indeed it is fixed. thanks
